### ISSUE TITLE
SonarQube not showing any issues

### ISSUE THREAD

**Issue by alexbde** [SonarOpenCommunity/sonar-cxx#812]
Hey guys, I'm following this project for a while because I got a problem with SonarQube not showing any issues in > 30k LOC. I know that's the goal of every developer but I'm quiet sure we haven't reached this point yet :wink: 

![sonarqube dashboard](https://cloud.githubusercontent.com/assets/17784818/13946959/21a5ceac-f018-11e5-8b1a-de5338b9eade.png)

As you can see other measures are calculated well like duplications or complexity. In _Components_ I'm also able to see all .cpp and .h files that matter.

I'm using `SonarQube 5.3` with `sonar-cxx-plugin-0.9.5-RC2`. I also tried it with `SonarQube 5.2` and `sonar-cxx-plugin-0.9.4` with same effect.

The analysis is run by Jenkins SonarQube runner as standalone analysis without previous build (because Jenkins is running on Unix machine and project is not capable building with Mono).

Have you got any idea how I can get SonarQube to calculate issues? Let me know if you need any further information.

**Comment by jmecosta** [comment]
hi,

i suppose you have not enable the rules from the cxx common repository in you quality profile. Once this is done, then you will see something. 

External tools like cppcheck require you to run them before the analysis. Check the wiki

If you use Windows, you can give a go to https://github.com/jmecsoftware/sonar-cxx-msbuild-tasks/releases

**Comment by alexbde** [comment]
This is an interesting hint! Which of these rules should I activate for default behaviour? Couldn't find something in the wiki.

![2016-03-22 11_51_52-2016-03-22 11_50_08-sonarqube png - fotos](https://cloud.githubusercontent.com/assets/17784818/13949300/81293498-f024-11e5-929a-188f92d5e02f.png)

**Comment by jmecosta** [comment]
C++ SonarQube C++ are the ones provided by the plugin.  I will had
reference to wiki if not there

**Comment by alexbde** [comment]
Thank you very much! I was able to fix my problem adding these rules and running `cppcheck` and `rats` before standalone SonarQube analysis.

To whom who may concern: add build step _execute shell script_ in Jenkins job config with following commands:

    rm -rf reports
    mkdir reports
    cppcheck --enable=all --inconclusive --xml --xml-version=2 ${WORKSPACE} 2> reports/cppcheck.xml
    rats --xml ${WORKSPACE} > reports/rats.xml

Tell SonarQube where reports are: set key `sonar.cxx.cppcheck.reportPath=reports/cppcheck.xml` and `sonar.cxx.rats.reportPath=reports/rats.xml` (via command line or config).

**Comment by palavaianoop** [comment]
same issue 
I am not sure what is the reason
![image](https://user-images.githubusercontent.com/29358331/43476645-5336a35c-94b6-11e8-9a2b-60b698a36123.png)
![image](https://user-images.githubusercontent.com/29358331/43476715-7fff109a-94b6-11e8-9166-1d829c74156b.png)

**Comment by guwirth** [comment]
Hi @palavaianoop,

* Please turn debug info on to get more info: https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Get-debug-information.
* Read also https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Path-and-path-separator-issues. Mostly a path issue in the reports.

Please create also a new issue and do not use a close one.

**Comment by palavaianoop** [comment]
Hi @guwirth 
weather, I have to install https://github.com/SonarQubeCommunity/sonar-cppcheck?

**Comment by guwirth** [comment]
@palavaianoop see 
https://github.com/SonarOpenCommunity/sonar-cxx/wiki/Running-the-analysis

**Comment by palavaianoop** [comment]
both are pretty much same!

**Comment by guwirth** [comment]
@palavaianoop please create a new issue and fill in the form. Thanks.
